ArduPlane/system.cpp

```diff
@@ -248,8 +248,12 @@ bool Plane::set_mode(Mode &new_mode, const ModeReason reason)
 
 #if AP_FENCE_ENABLED
     // may not be allowed to change mode if recovering from fence breach
-    if (hal.util->get_soft_armed() && fence.enabled() && fence.option_enabled(AC_Fence::OPTIONS::DISABLE_MODE_CHANGE) &&
-                                            fence.get_breaches() && in_fence_recovery()) {
+    if (hal.util->get_soft_armed() &&
+        fence.enabled() &&
+        fence.option_enabled(AC_Fence::OPTIONS::DISABLE_MODE_CHANGE) &&
+        fence.get_breaches() &&
+        in_fence_recovery() &&
+        reason != ModeReason::RTL_COMPLETE_SWITCHING_TO_FIXEDWING_AUTOLAND) {
```

Comment:
Should include these other cases too:
https://github.com/ArduPilot/ardupilot/blob/71fdf37055ea18e83635292c6102a61834e98d02/ArduPlane/fence.cpp#L141-L144

Reply:
you're right, thanks, PR here: https://github.com/ArduPilot/ardupilot/pull/22041